### DISCUSSION TITLE
Send requests to /v1/completions return {"detail":"Not Found"}

### DISCUSSION THREAD

**Discussion by unixshaman** [open-webui/open-webui#5049] [General]
# Bug Report

## Installation Method

Docker

## Environment

- **Open WebUI Version:** v0.3.16
- **Ollama (if applicable):** 0.3.6

- **Operating System:** Ubuntu 22.04
- **Browser (if applicable):** api client

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

I need api /v1/completions like described here https://github.com/ollama/ollama/blob/main/docs/openai.md#v1completions.
When execute request,  get response like from original ollama:
`{"id":"cmpl-648","object":"text_completion","created":1725031280,"model":"llama3.1:8b","system_fingerprint":"fp_ollama","choices":[{"text":"I'm ready for the test. Go ahead and administer it. I'll do my best to respond appropriately. What's the first question?","index":0,"finish_reason":"stop"}],"usage":{"prompt_tokens":16,"completion_tokens":29,"total_tokens":45}}
`

## Actual Behavior:

I need api /v1/completions like described here https://github.com/ollama/ollama/blob/main/docs/openai.md#v1completions.
When execute request,  get {"detail":"Not Found"}.

## Description

**Bug Summary:**
Method /v1/completions send wrong response.

## Reproduction Details

**Steps to Reproduce:**
Send request:
`
POST https://host.com/ollama/v1/completions HTTP/1.1
Content-Type: application/json
Authorization: Bearer sk-*****
Host: host.com
Content-Length: 69

{
    "model": "llama3.1:8b",
    "prompt": "Say this is a test"
}
`

## Logs and Screenshots

**Docker Container Logs:**
INFO:     188.130.248.110:0 - "POST /ollama/v1/completions HTTP/1.0" 404 Not Found

**Screenshots/Screen Recordings (if applicable):**
![image](https://github.com/user-attachments/assets/3c56ac61-aa0d-4751-8df5-7a4dde3e3623)

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by unixshaman** [comment]
I really hope that this is indeed a bug and not some kind of carelessness on my part.

**Reply by tjbck** [reply]
`/v1/chat/completions` :)

**Reply by unixshaman** [reply]
But /v1/chat/completions and /v1/completions different things ?
https://stackoverflow.com/questions/76192496/openai-v1-completions-vs-v1-chat-completions-end-points may be here found some info.
Isn't that right?

Ollama support both endpoints.
I need it, because https://github.com/continuedev/continue using this api endpoint. Log from Continue plugin bottom.
![image](https://github.com/user-attachments/assets/664fcaba-77a4-4fde-8114-b32265159475)

**Reply by unixshaman** [reply]
> `/v1/chat/completions` :)

What about my answer?

**Comment by Ethyk** [comment]
I had the same problem as you and this link help me to fix it.
I added this line `"useLegacyCompletionsEndpoint": false` in `tabAutocompleteModel`

https://github.com/continuedev/continue/issues/1894#issuecomment-2265772202

hope this helps you

**Reply by unixshaman** [reply]
Yes, you are absolutely right, it works. Thank you very much for the tip.

But still, it is strange that the original Ollama supports both points, and open-webui does not proxy requests.

**Reply by unixshaman** [reply]
@Ethyk have you tried the plugin https://github.com/carlrobertoh/CodeGPT and used it with open-web recently?

**Comment by unixshaman** [comment]
Returning to this question:
ollama supports API /api/completion, vllm supports API /api/completion, users often use them for autocomplete tasks in IDE plugins. But now, for this to work, we have to make a separate service, and on the open-webui side use pipelines and a bunch of other crutches.

Yes, I know that for OpenAI /api/completionon is considered legacy, but such models as qwen coder, deepseek coder and others use it.
Why can't open-webui support this API?

**Reply by unixshaman** [reply]
@tjbck Please, give some feedback. Thank you very much!

**Reply by tjbck** [reply]
Support added to dev!

**Reply by unixshaman** [reply]
@tjbck This is awesome! Thank you very much for the implementation.
I will try to run from the dev branch, but I am not so fast ))
I want to check if it will work with vllm, which is connected via openai.

**Reply by AlexanderZhk** [reply]
> Support added to dev!

Awesome work! Any chance to see support for the openai app and the webui api in general? Would be very useful, when using openwebui as an IDE autocomplete "model-switching" backend.

Do you think opening an issue would be appropriate for this? Surely it's popular and there are more use-cases for this, there was quite the pushback on the oai forums, when they moved `/completion` to legacy, as it is very useful for a lot of tasks.

**Reply by mennohofste** [reply]
What happened to these changes? They do not appear to be in dev anymore, but `main` does not seem to have this.

I am running into the same issue, and `useLegacyCompletionsEndpoint` is not really an option for me. It gives bogus results with the model I am using.

Wondering if anyone else has some kind of solution they are willing to share.
